lib/twilio-ruby/http/http_client.rb

```diff
@@ -31,7 +31,11 @@ def _request(request)
           f.options.params_encoder = Faraday::FlatParamsEncoder
           f.request :url_encoded
           f.headers = request.headers
-          f.request(:basic_auth, request.auth[0], request.auth[1])
+          if Faraday::VERSION =~ /^2/
```

Comment:
`String#start_with?` should be faster than using a regular expression.
```suggestion
          if Faraday::VERSION.start_with?('2.')
```

Reply:
Could also use `Gem::Version` which is a common pattern in ruby gems. Whatever you all prefer, I will update.

```suggestion
          if Gem::Version.new(Faraday::VERSION) >= Gem::Version.new("2")
```

Reply:
Since this code is evaluated with every request, I think that it's better to avoid creating new objects.